docs/reference/api/connectivity/drivers/EMACMemoryManager.md

```diff
@@ -0,0 +1,24 @@
+## EMAC memory manager
+
+The Ethernet MAC memory manager class provides abstracted memory interface toward memory modules used in different network stacks. The EMAC device driver uses the memory interface to handle memory related operations, such as allocation, free and data manipulation. The [EMAC](emac.html) drivers use this memory manager to reserve space for network packets.
+
+Memory buffer chains store the data on the memory interface. The data passed in either direction either may be contiguous (a single-buffer chain) or may consist of multiple buffers. You can chain the buffers using a singly linked list.
+
+### Usage
+
+The EMAC driver can allocate memory either from a memory pool or from the heap (contiguous memory). By preference, link input memory should be allocated from the pool, but if contiguous memory is required by the hardware, it can be allocated from the heap.
+
+The driver can express alignment preferences for buffers sent to it, but the network stack is not required to meet these preferences. This means that the drivers relying on alignment need to check the buffer address and may need to copy the output data into an aligned (or contiguous) buffer when its requirements are not met.
```

Comment:
Query: Is this saying that drivers the rely on alignment must check the buffer address, and those drivers may need to copy the output data into an aligned (or contiguous) buffer when the buffer address's needs aren't met?

Reply:
Yes, correct.
Feel free to change the messaging if you feel that it is not expressing its intent clearly enough.